x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java

```diff
@@ -74,21 +75,23 @@
 import static org.elasticsearch.xpack.application.connector.ConnectorFiltering.fromXContentBytesConnectorFiltering;
 import static org.elasticsearch.xpack.application.connector.ConnectorFiltering.sortFilteringRulesByOrder;
 import static org.elasticsearch.xpack.application.connector.ConnectorTemplateRegistry.MANAGED_CONNECTOR_INDEX_PREFIX;
+import static org.elasticsearch.xpack.core.ClientHelper.CONNECTORS_ORIGIN;
 
 /**
  * A service that manages persistent {@link Connector} configurations.
  */
 public class ConnectorIndexService {
 
-    private final Client client;
+    // The client to interact with the system index (internal user).
+    private final Client clientWithOrigin;
 
     public static final String CONNECTOR_INDEX_NAME = ConnectorTemplateRegistry.CONNECTOR_INDEX_NAME_PATTERN;
 
     /**
      * @param client A client for executing actions on the connector index
      */
     public ConnectorIndexService(Client client) {
-        this.client = client;
+        this.clientWithOrigin = new OriginSettingClient(client, CONNECTORS_ORIGIN);
```

Comment:
if you make this variable reuse the old name, a LOT of the diff goes away

```
this.client = new OriginSettingClient(client, CONNECTORS_ORIGIN);
```

Reply:
(same note for the ConnectorSyncJobIndexService.java)

Reply:
I feel like naming `OriginSettingClient` as `clientWithOrigin` is established patter in ES https://github.com/search?q=repo%3Aelastic%2Felasticsearch+clientWithOrigin&type=code therefore I prefer to keep it like this so that we are explicit how we access the soon to be  system indices